### PR TITLE
Allow to remove an admin extension dynamically

### PR DESCRIPTION
**Pull request by VincentLanglet**
## Subject

I am targeting this branch, because this is not possible in 4.x since getList is final and this will allow me to solve deprecation to update from 3.x to 4.x.

## Changelog

```markdown
### Added
- AbstractAdmin::removeExtension()
```

### CODE REVIEW

File: src/Admin/AbstractAdmin.php [hunk 1]
```diff
@@ -2937,6 +2937,18 @@ public function addExtension(AdminExtensionInterface $extension)
         $this->extensions[] = $extension;
     }
 
+    final public function removeExtension(AdminExtensionInterface $extension): void
```
Comment (core23) [inline src/Admin/AbstractAdmin.php:2940]: Will this work when using the `ExtensionCompilerPass`?

Can you give an example how and when the `removeExtension` can be called?

Comment (VincentLanglet) [reply src/Admin/AbstractAdmin.php:2940]:
```
public function configure()
{
     if ($someCondition) {
          foreach ($this->getExtensions() as $extension) {
              if ($extension instanceof Foo) {
                  $this->removeExtension($extension);
              }
          }
    }
}
```

I tested on my project with success.

**Review by core23 — DISMISSED** [review]

File: src/Admin/AbstractAdmin.php (hunk 1 above)
Comment (jordisala1991) [reply src/Admin/AbstractAdmin.php:2940]: IMO we should provide some docs about this feature

File: tests/Admin/AdminTest.php [hunk 2]
```diff
@@ -948,6 +948,9 @@ public function testGetExtensions(): void
         $admin->addExtension($adminExtension1);
         $admin->addExtension($adminExtension2);
         static::assertSame([$adminExtension1, $adminExtension2], $admin->getExtensions());
+
+        $admin->removeExtension($adminExtension2);
+        static::assertSame([$adminExtension1], $admin->getExtensions());
```
Comment (jordisala1991) [inline tests/Admin/AdminTest.php:953]: Would be nice if we also check for removing non existent extension

Comment (VincentLanglet) [reply tests/Admin/AdminTest.php:953]: Done

File: src/Admin/AbstractAdmin.php (hunk 1 above)
Comment (VincentLanglet) [reply src/Admin/AbstractAdmin.php:2940]: Done

**Review by core23 — APPROVED** [review]

**Review by jordisala1991 — APPROVED** [review]

**Pull request merged**